packages/medusa/src/types/customers.ts

```diff
@@ -1,11 +1,17 @@
-import { IsOptional, IsString } from "class-validator"
+import { Transform } from "class-transformer"
+import { IsBoolean, IsOptional, IsString } from "class-validator"
 import { AddressPayload } from "./common"
 
 export class AdminListCustomerSelector {
   @IsString()
   @IsOptional()
   q?: string
 
+  @IsBoolean()
+  @IsOptional()
+  @Transform(({ value }) => value === "true")
```

Comment:
todo: Use our custom boolean validator
```suggestion
  @Transform(({ value }) => optionalBooleanMapper.get(value))
```